data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/configuration/KinesisStreamConfig.java

```diff
@@ -0,0 +1,32 @@
+package org.opensearch.dataprepper.plugins.kinesis.source.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import software.amazon.kinesis.common.InitialPositionInStream;
+
+@Getter
+public class KinesisStreamConfig {
+    // Checkpointing interval
+    private static final int MINIMAL_CHECKPOINT_INTERVAL_MILLIS = 2 * 60 * 1000; // 2 minute
+    private static final boolean DEFAULT_ENABLE_CHECKPOINT = false;
+
+    @JsonProperty("stream_name")
+    @NotNull
+    @Valid
+    private String name;
+
+    @JsonProperty("stream_arn")
+    private String arn;
```

Comment:
Why do we provide both a stream_name and a stream_arn?

Reply:
`stream_arn` is not required. This attribute is being derived from the `StreamDescription`. I have removed this.